### PR TITLE
feat: create cover sheet for printing (FPLA-1086)

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [x] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1086

### Change description ###
- Adds the 'cover sheet' Docmosis template.
- Adds DocmosisCoverGeneratorService that exposes a method to build the cover sheet. Package private methods also included to build template data relevant to the template.

[cover-sheet-sample.pdf](https://github.com/hmcts/fpl-ccd-configuration/files/4133788/care_order-arial.pdf)

This will be tested properly in #826 when full integration with system is available

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java [hunk 1]
```diff
@@ -165,6 +169,6 @@ private String concatGatewayConfigurationUrlAndMostRecentUploadedOrderDocumentPa
     }
 
     private DocmosisTemplates getDocmosisTemplateType(GeneratedOrderType type) {
-        return type == EMERGENCY_PROTECTION_ORDER ? EPO : ORDER;
+        return type == EMERGENCY_PROTECTION_ORDER ? EPO : COVER_SHEET;
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java]: Hack - Will be reverted. Using the orders mid event to generate the COVER_SHEET for QA.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ private Document getDocument(String authorization,
         DocmosisTemplates templateType = getDocmosisTemplateType(caseData.getOrderTypeAndDocument().getType());
 
         DocmosisDocument document = docmosisDocumentGeneratorService.generateDocmosisDocument(
-                service.getOrderTemplateData(caseData), templateType);
+            service.getOrderTemplateData(caseData), templateType);
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java]: 👀

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 3]
```diff
@@ -20,6 +20,7 @@
 import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
 import uk.gov.hmcts.reform.fpl.model.emergencyprotectionorder.EPOChildren;
 import uk.gov.hmcts.reform.fpl.model.emergencyprotectionorder.EPOPhrase;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java]: 👀

**Comment by Jamesnelson117** [comment]
@lewisbirks Thank you for your cherished approval.

**Pull request closed without merging**